openapi-model/src/main/scala/sttp/apispec/openapi/validation/OpenAPIComparator.scala

```diff
@@ -0,0 +1,277 @@
+package sttp.apispec.openapi.validation
+
+import sttp.apispec.{Schema, SchemaLike}
+import sttp.apispec.openapi.{Header, MediaType, OpenAPI, Operation, Parameter, PathItem, RequestBody, Response}
+import sttp.apispec.validation.SchemaComparator
+
+import scala.collection.immutable.ListMap
+
+class OpenAPIComparator(
+    clientOpenAPI: OpenAPI,
+    serverOpenAPI: OpenAPI
+) {
+
+  private val httpMethods = List(
+    ("get", (_: PathItem).get),
+    ("post", (_: PathItem).post),
+    ("patch", (_: PathItem).patch),
+    ("delete", (_: PathItem).delete),
+    ("options", (_: PathItem).options),
+    ("trace", (_: PathItem).trace),
+    ("head", (_: PathItem).head),
+    ("put", (_: PathItem).put)
+  )
+
+  private val clientSchemas: Map[String, Schema] = clientOpenAPI.components match {
+    case Some(components) =>
+      components.schemas.flatMap {
+        case (key, schema: Schema) => Some((key, schema))
+        case _                     => None
+      }
+    case _ => Map.empty[String, Schema]
+  }
+
+  private val serverSchemas: Map[String, Schema] = serverOpenAPI.components match {
+    case Some(components) =>
+      components.schemas.flatMap {
+        case (key, schema: Schema) => Some((key, schema))
+        case _                     => None
+      }
+    case _ => Map.empty[String, Schema]
+  }
+
+  def compare(): List[OpenAPICompatibilityIssue] = {
```

Comment:
Some docs here would be nice, specifying the extent of the comparison - paths, schemas, etc., and what is not compared. From the perspective of a first-time user